central/virtualmachine/service/service_impl.go

```diff
@@ -76,6 +94,25 @@ func (s *serviceImpl) ListVirtualMachines(ctx context.Context, request *v2.ListV
 	for _, vm := range vms {
 		v2VMs = append(v2VMs, storagetov2.VirtualMachine(vm))
 	}
+	requestQueryPagination := request.GetQuery().GetPagination()
+	if requestQueryPagination.GetSortOption() == nil && len(requestQueryPagination.GetSortOptions()) == 0 {
```

Comment:
Sorting in go will likely mess up paging.  Take a look at https://github.com/stackrox/stackrox/blob/1ff122adf24e7f828949c76a3a969f26e303b099/central/policy/store/postgres/gen.go#L3
Focus on the Default sort option that I added to store generation when I got rid of the searchers.  Then you can put it on the query assuming the sort fields are search fields.  What I don't remember is if I handled multiple fields or just one.